### PR TITLE
Add yolov8n tutorial with integrated post processing

### PR DESCRIPTION
**Pull request by Idan-BenAmi**
## Pull Request Description:
In this pull request I added tutorial of yolov8n object detection model with integrated post processing. I also align nanodet tutorial to work with the same utility functions (remove duplications) and to work with google colab.

## Checklist before requesting a review:
- [x] I set the appropriate labels on the pull request.
- [x] I have added/updated the release note draft (if necessary).
- [x] I have updated the documentation to reflect my changes (if necessary).
- [x] All function and files are well documented. 
- [x] All function and classes have type hints.
- [x] There is a licenses in all file.
- [x] The function and variable names are informative. 
- [x] I have checked for code duplications.
- [x] I have added new unittest (if necessary).

### CODE REVIEW

File: tutorials/notebooks/example_keras_yolov8n.ipynb [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+    "\n",
+    "    def representative_dataset():\n",
+    "        ds_iter = iter(train_loader)\n",
+    "        for _ in range(n_iter):\n",
+    "            yield [next(ds_iter)[0]]\n",
+    "\n",
+    "    return representative_dataset\n",
+    "\n",
+    "# Preform post training quantization \n",
+    "quant_model, _ = mct.ptq.keras_post_training_quantization_experimental(model,\n",
```
Comment (elad-c) [inline tutorials/notebooks/example_keras_yolov8n.ipynb:284]: if this model should run on the IMX500, define the explicit TPC:
```
# Set IMX500-v1 TPC
tpc = mct.get_target_platform_capabilities("tensorflow", 'imx500', target_platform_version='v1')
```

File: tutorials/notebooks/example_keras_yolov8n.ipynb [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+    "print('Quantized model is ready')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4fb6bffc-23d1-4852-8ec5-9007361c8eeb",
+   "metadata": {},
+   "source": [
+    "### Evaluate quantized model\n",
+    "Lastly, we can evaluate the performance of the quantized model. There is a slight decrease in performance that can be further mitigated by either expanding the representative dataset or employing MCT's advanced quantization methods, such as EPTQ (Enhanced Post Training Quantization)."
```
Comment (elad-c) [inline tutorials/notebooks/example_keras_yolov8n.ipynb:296]: still called GPTQ in the MCT :)

File: tutorials/resources/nanodet/__init__.py
Comment (elad-c) [inline tutorials/resources/nanodet/__init__.py:1]: Add license

File: tutorials/resources/yolov8/__init__.py
Comment (elad-c) [inline tutorials/resources/yolov8/__init__.py:1]: license

File: tutorials/resources/yolov8/yolov8_keras.py [hunk 3]
```diff
@@ -0,0 +1,516 @@
+# Copyright 2023 Sony Semiconductor Israel, Inc. All rights reserved.
```
Comment (elad-c) [inline tutorials/resources/yolov8/yolov8_keras.py:1]: if this code was copied from ultralytics and modified, then you need to change the license. Check out the license in

> https://github.com/sony/model_optimization/blob/main/tutorials/resources/efficientdet/effdet_keras.py

File: tutorials/resources/yolov8/yolov8_keras.py [hunk 4]
```diff
@@ -0,0 +1,516 @@
...
+        max_total_size=300,
+        iou_threshold=0.7,
+        score_threshold=0.001,
+        pad_per_class=False,
+        clip_boxes=False
+        )
+
+    model = Model(model.input, outputs, name='yolov8n')
+
+    model.save('/data/projects/swat/network_database/Tensorflow2/float/yolov8n/temp.h5')
```
Comment (elad-c) [inline tutorials/resources/yolov8/yolov8_keras.py:515]: remove this

File: tutorials/resources/yolov8/yolov8_keras.py [hunk 5]
```diff
@@ -0,0 +1,516 @@
...
+    )
+
+    x = np.flip(x, -1) # Flip image channels
+    x = cv2.resize(x, (wr, hr), interpolation=cv2.INTER_AREA) # Aspect ratio preserving resize
+    x = np.pad(x, pad, constant_values=pad_values) # Padding to the target size
+    x = (x - img_mean) / img_std # Normalization
+    return x
+
+
+if __name__ == '__main__':
```
Comment (elad-c) [inline tutorials/resources/yolov8/yolov8_keras.py:485]: remove this

File: tutorials/resources/yolov8/yolov8n.yaml [hunk 6]
```diff
@@ -0,0 +1,59 @@
+# Copyright 2023 Sony Semiconductor Israel, Inc. All rights reserved.
```
Comment (elad-c) [inline tutorials/resources/yolov8/yolov8n.yaml:1]: Same comment about license. if you didn't change the file at all, maybe download it online

**Review by elad-c — CHANGES_REQUESTED** [review]

File: tutorials/notebooks/example_keras_yolov8n.ipynb [hunk 7]
```diff
@@ -0,0 +1,351 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e8395b28-4732-4d18-b081-5d3bdf508691",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!wget -nc https://github.com/ultralytics/assets/releases/download/v0.0.0/yolov8n.pt\n",
+    "import torch\n",
+    "\n",
+    "PRETRAINED_WEIGHTS_FILE = '/content/yolov8n.pt'\n",
```
[the remark this answers is not in the document]
Comment (Idan-BenAmi) [reply tutorials/notebooks/example_keras_yolov8n.ipynb:128]: I keep it to emphasize this path could be defined by the user (depends if he uses collab or his local disk)

File: tutorials/notebooks/example_keras_yolov8n.ipynb [hunk 8]
```diff
@@ -0,0 +1,351 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3cde2f8e-0642-4374-a1f4-df2775fe7767",
+   "metadata": {},
+   "source": [
+    "#### Evaluate the floating point model\n",
+    "Next, we evaluate the floating point model by using `cocoeval` library alongside additional dataset utilities. We can verify the mAP accuracy aligns with that of the original model. \n",
+    "Note that we set the \"batch_size\" to 5 and the preprocessing according to [Ultralytics](https://github.com/ultralytics/ultralytics).\n",
+    "Please ensure that the dataset path has been set correctly before running this code cell."
```
[the remark this answers is not in the document]
Comment (Idan-BenAmi) [reply tutorials/notebooks/example_keras_yolov8n.ipynb:195]: Again, just to emphasize this could be modified if needed.

File: tutorials/notebooks/example_keras_nanodet_plus.ipynb [hunk 9]
```diff
@@ -266,30 +266,31 @@
...
+    "                                                batch_size=BATCH_SIZE)\n",
     "\n",
     "# Define representative dataset generator\n",
-    "def get_representative_dataset(n_iter, train_loader):\n",
-    "\n",
-    "    def representative_dataset():\n",
-    "        ds_iter = iter(train_loader)\n",
+    "def get_representative_dataset(n_iter: int, dataset_loader: Iterator[Tuple]):\n",
+    "    \n",
+    "    def representative_dataset() -> Iterator[List]:\n",
```
Comment (elad-c) [inline tutorials/notebooks/example_keras_nanodet_plus.ipynb:284]: Need function doc:

>  
       """
        Creates a representative dataset generator from a PyTorch data loader, The generator yields numpy
        arrays of batches of shape: [Batch, H, W ,C]
        Returns:
            A representative dataset generator
        """

**Review by elad-c — APPROVED** [review]
checkout the comments
